interview-questions.md

```diff
@@ -8,50 +8,123 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+
+Source:
+
+    https://thesassway.com/what-is-an-index-in-computer-science/#:~:text=in%20a%20database%3F-,What%20Is%20An%20Index%20In%20Computer%20Programming%3F,any%20arbitrary%20sequence%20of%20values.
+    https://www.computerhope.com/jargon/v/value.htm#:~:text=In%20mathematics%2C%20logic%2C%20and%20computer,or%20a%20string%20of%20characters.
+
+    
+
+
 4. What is iteration?
 
 Your answer:
 
+ Iteration is like the energizer bunny it allows us to simplfy mathamatical equasion by repeating certain steps until told otherwise.
+
 Researched answer:
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-Your answer:
+Your answer: 
+
+    A good description of proper pair programming techniques includes taking turns, it gives you the opportunity to problem solvewhile yout partner learns typing syntax and vice verse-a. Pair programming also helps reduce the number of mistakes made. For example sometimes you can be typing and miss spell a file name your partner will be there to navigate you. Most of the time they correct the mistake before you even finish you line of code. In pair programinf one person is the Navigator and the other perdon is the Driver.
+
+Researched answer: 
+
+Why We Utilize Pair Programming
+
+Different people have different knowledge and strengths which can lead to better solutions
+Distributing cognitive load and use individualized skills (typing syntax vs. problem solving) when driving and navigating
+Coding can feel overwhelming, so focusing on one aspect of the job at hand can help focus energy on problem solving
+Two sets of eyes makes it easier to catch mistakes and produce high-quality code
+A pair can hold each other accountable so you'll be less likely to be tempted to take shortcuts or jump around without finishing the task you are on
+Partners help keep the focus on coding and not on distractions like texts, emails, etc
+Talking about your code with a partner will deepen your understanding of programming
+Increased specialization requires more collaboration
+
+Roles: Driver and Navigator
+The driver is the one with their hands on the keyboard. They are responsible for entering information and following the instructions given by the navigator. This approach requires constant communication. And talking about your code will make you a better developer and a better teammate. It is important to switch roles every 10-20 minutes and take frequent breaks.
+
+
+Source: 
+
+   https://github.com/learn-academy-2022-echo/Syllabus/blob/main/tools-and-resources/pair-programming.md
+
+
 
-Researched answer:
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
```

Comment:
It's always great to look ahead at the terms so you have some context when hearing the terms during lectures.